django/contrib/postgres/fields/array.py

```diff
@@ -298,15 +304,18 @@ def __call__(self, *args, **kwargs):
 
 
 class SliceTransform(Transform):
-
     def __init__(self, start, end, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.start = start
         self.end = end
 
     def as_sql(self, compiler, connection):
         lhs, params = compiler.compile(self.lhs)
-        return '%s[%%s:%%s]' % lhs, params + [self.start, self.end]
+        # self.start is set to 1 if slice start is not provided.
+        if self.end:
+            return f'{lhs}[%s:%s]', params + [self.start, self.end]
+        else:
+            return f'{lhs}[%s:]', params + [self.start]
```

Comment:
If changing to `None` from `''` in `.slice_expression()` above, then:

```suggestion
        if self.end is None:
            return f'{lhs}[%s:]', params + [self.start]
        else:
            return f'{lhs}[%s:%s]', params + [self.start, self.end]
```